### DISCUSSION TITLE
Soo. now what, how do I <insert subject> (any subject cause I am lost)

### DISCUSSION THREAD

**Discussion by DrLogistik** [thehighnibble/thehighnibble.github.io#152] [IMSAI8080esp replica]
Hello all, 
So I just finished putting this marvelous thing together, and I managed to get it to connect to wifi.  I am able to load CP/M per the videos that I watched. but now what? 

I do not know how to use CPM? 
I do not know how and where and what to explore? 

I really can't figure out how to do anything but use the TTY and get confused by my lack of knowledge.

IS there any type of manual, or newbie documentation on how to play around and do things.  

OH. I was able to load into BASIC (again in the videos) and make a hello world program, so I have done something :) 

anyways, any help would be appreciated. 

Someone give me a homework assignment on what to do and how to start playing around.  
please

I am so lost..

All I want to figure out is how to load one of the tape files first :) 

I did skim through the MAN: folder on the desktop UI, but it just confused me more.

**Comment by nbreeden2** [comment]
An original IMSAI back in the day was a blue box with blinking lights, switches and a manual. There was very little else to act as a guide. Things like reading a paper tape into memory took a lot of reading and if you were lucky having a local user's group or someone who had one. You're right at that point now just as it was in 1977, What do I do next? Part of the experience of the emulator is the need to figure things out just as those of us who built computers in the 70s had too.

You're correct that you need to get comfortable with CPM and the command line interface for it before you can do a lot else. There should be a "CPM Users" manual in the MAN: folder that covers CPM and its command line interface.

CPM is strictly a command line interface. A keyboard and screen. Everything else you see in the WebUI are the emulated peripherals etc. that the emulator can use. CPM has no UI, no graphical interface. In the real world you would plug in a DAZZLER card, configure it, plug in a TV and hope the software did something, assuming you could read the 8-inch disk. The High Nibble has hidden much of this detail work in the WebUI, the WebSocket control button on each emulated peripheral does this magic for you.

This video demonstrates how to load paper tape files for a specific use case.
https://www.youtube.com/watch?v=p3aPnFFPc6w

I know of no generic "how to load a paper-tape file" document or demonstration. This is something I might consider creating. 

To really get under the covers you need to understand things like ROMs and how they are connected inside the emulation. A hex file represents a ROM. In a real system you had to plug it into a card, configure the card etc. then hope it worked. Much of this work is captured in the core emulator configuration and the config files you can access from the WebUI.

You'll also find manuals and other CPM 2.2 materials at: 
http://www.cpm.z80.de/drilib.html
Note that a lot of the web site covers other versions of CP/M so be sure to find the sections that apply to 2.2
